scipy/stats/_qmc.py

```diff
@@ -1141,8 +1187,13 @@ class MultivariateNormalQMC(QMCEngine):
 
     """
 
-    def __init__(self, mean, cov=None, cov_root=None, inv_transform=True,
-                 engine=None, seed=None):
+    def __init__(
+            self, mean: np.typing.ArrayLike, cov: Optional[np.typing.ArrayLike] = None,
```

Comment:
For example here. I have no idea what is happening here and I don't think any IDE can help me out here to understand what the signature is. 

If I remember correctly for keywords it will automatically register as Optional.

Reply:
This is ok for an IDE. We need the `Optional` for mypy as it's not enough that the default is `None`. If that was your concern.

Reply:
Shouldn't this be eligible for https://mypy.readthedocs.io/en/stable/runtime_troubles.html#using-x-y-syntax-for-unions ?

Reply:
So we can write `ArrayLike | None` maybe instead of this garbled stuff?

Reply:
Union? I think there might be a misunderstanding here. There are 2 parameters here `mean` and `cov`

Reply:
`Optional[npt.ArrayLike]` or without the `npt` but not sure about the `|`. Is this supposed to replace the Optional too?

Reply:
Apparently yes (https://www.python.org/dev/peps/pep-0604/#proposal). Not sure about the best practices we want here. But as we will still need to write the default for python itself, IMO this will look redundant as `ArrayLike | None = None`

Reply:
Didn't change it then